plugins/mbpi.c

```diff
@@ -565,34 +564,39 @@ static gboolean mbpi_parse(const GMarkupParser *parser, gpointer userdata,
 	char *db;
 	int fd;
 	GMarkupParseContext *context;
-	gboolean ret;
+	gboolean ret = FALSE;
+	const char *snap;
+	char *filename;
+
+	snap = getenv("SNAP");
+	filename = g_strdup_printf("%s%s", snap ? snap : "", MBPI_DATABASE);
```

Comment:
We should discuss where mbpi ends up living as it's used by modemmanager too.

I think it probably should live in it's own snap that shares the file via content sharing.  We also should explore the ability for an end-user to modify this file on a working system ( ie. allow a writable version that would override the default ) vs. forcing the user to report a bug and wait for an official update.